aopy/preproc/wrappers.py

```diff
@@ -370,3 +384,53 @@ def proc_lfp(data_dir, files, result_dir, result_filename, overwrite=False, max_
                                                 max_memory_gb=max_memory_gb, **filter_kwargs)
         
     aodata.save_hdf(result_dir, result_filename, lfp_metadata, "/lfp_metadata", append=True)
+
+
+
+def proc_emg(data_dir, files, result_dir, result_filename, overwrite=False, max_memory_gb=1., filter_kwargs={}):
+    '''
+    Process emg data:
+        Loads 'broadband' hdf data or 'ecube' headstage data and metadata
+    Saves lfp data into the HDF datasets:
+        lfp_data (nt, nch)
+        lfp_metadata (dict)
+    
+    Args:
+        data_dir (str): where the data files are located
+        files (dict): dictionary of filenames indexed by system
+        result_filename (str): where to store the processed result
+        overwrite (bool, optional): whether to remove existing processed files if they exist
+        max_memory_gb (float, optional): max memory used to load binary data at one time
+        filter_kwargs (dict, optional): keyword arguments to pass to :func:`aopy.precondition.filter_lfp`
+    '''  
+    # Check if a processed file already exists
+    filepath = os.path.join(result_dir, result_filename)
+    if not overwrite and os.path.exists(filepath):
+        contents = aodata.get_hdf_dictionary(result_dir, result_filename)
+        if "emg" in contents:
+            print("File {} already preprocessed, doing nothing.".format(result_filename))
+            return
+    elif os.path.exists(filepath):
+        os.remove(filepath)
+
+    # Preprocess neural data into lfp   
+    
+    if 'emg' in files:
+        result_filepath = os.path.join(result_dir, result_filename)
+        
+        emg_data, emg_metadata = aodata.load_bmi3d_hdf_table(data_dir, files['emg'], 'data')
+        #Channels 0:64 Are EMG
+        #Need to add functionality to deal with different shaped EMG inputs
+        emg_data_reshape = emg_data.view(('f8', (len(emg_data.dtype),)))
+    
+        clck = emg_data_reshape[:,-24] # I think this should always grab the clock signal
+
+        clck = np.round(clck/30000) # This is hardcoded currently based on
+
+        ts, values = detect_edges(clck>0.8, 2048, rising=True, falling=False) # Currently undoing the time transform because Quatt sampling rate is unreliable
+        emg_clck_idx = (ts*2048).astype(int)
```

Comment:
this logic is already covered by aopy.preproc.bmi3d.parse_bmi3d, in theory. we might need to play around with some metadata setting to get it to work but no need to repeat it here